### PR TITLE
feat(finetune): add adapter-only output

### PR DESCRIPTION
**Pull request by sozercan**
**What this PR does / why we need it**:

Adds `output.format: adapter` so fine-tuning can publish a portable PEFT LoRA adapter bundle instead of always converting to GGUF.

- Keeps `gguf` as the default for backward compatibility.
- Rejects adapter output combined with quantization.
- Validates and stages adapter weights, configuration, tokenizer assets, and integrity metadata without base-model weights.
- Skips the GGUF export phase and preserves the training cache across output formats.
- Adds unit coverage, documentation, and an `adapter-smoke` GPU workflow mode.

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:

None.

**Special notes for your reviewer**:

Adapter bundles require the referenced base weights and a compatible PEFT/Transformers consumer. Existing AIKit runners do not directly serve this output format.

Local validation:
- `go test ./...`
- Focused Go race tests
- Python fine-tune tests
- Docusaurus production build
- `actionlint` (apart from the repository-specific self-hosted `gpu` label warning)

### CODE REVIEW

File: pkg/finetune/target_unsloth.py [hunk 1]
```diff
@@ -4553,11 +4678,20 @@ def train_model(
                 )
             source_datasets.append(source_dataset)
 
+    training_model_name, training_model_revision = resolve_model_snapshot(
+        train_config["baseModel"],
+        load_in_4bit=cfg["loadIn4bit"],
+        description="resolved training model revision",
+        model_info=dependencies.model_info,
+        resolve_model_name=dependencies.resolve_model_name,
+    )
```
[the remark this answers is not in the document]
Comment (sozercan) [reply pkg/finetune/target_unsloth.py:5501]: Fixed in 2992e83. Training now resolves one immutable repository/SHA using the configured loadIn4bit setting, passes that exact tuple to from_pretrained with use_exact_model_name=True, and records the same tuple in the PEFT config. The independent non-4-bit lookup was removed, so adapter consumers and GGUF export no longer use unverifiable provenance. Tests assert the load and published config use the same repository and revision.

File: pkg/finetune/target_unsloth.py [hunk 2]
```diff
@@ -480,6 +500,104 @@ def pin_peft_base_model(
...
+        filename = artifact_path.name
+        if (
+            filename == "adapter_model.bin"
+            or filename == "config.json"
+            or filename.endswith(".gguf")
+            or filename.startswith("pytorch_model")
+            or (
+                filename.startswith("model")
+                and filename.endswith(".safetensors")
+            )
```
[the remark this answers is not in the document]
Comment (sozercan) [reply pkg/finetune/target_unsloth.py:624]: Fixed in 2992e83. Validation now permits adapter_model.safetensors only at the bundle root and rejects model-weight suffixes across every Path suffix, including nested weights and compound checkpoint/index names. Tests cover the reported h5, msgpack, pth, and nested adapter cases plus .bin.index.json, .safetensors.index.json, .msgpack.index.json, and .ckpt index/data variants. The GPU workflow mirrors these checks.

File: website/docs/specs-finetune.md [hunk 3]
```diff
@@ -39,10 +39,35 @@ config:
...
+```text
+<name>/
+├── adapter_config.json
+├── adapter_model.safetensors
+├── tokenizer_config.json
+├── tokenizer.json (or model-specific tokenizer vocabulary files)
+└── optional tokenizer assets and templates
+```
+
+The bundle does not contain base-model weights or a GGUF file. `adapter_config.json` identifies the resolved base-model repository and immutable revision used for training. A consumer must load that compatible base model, apply the adapter with a PEFT-compatible runtime, and use the bundled tokenizer configuration and templates. The adapter directory is not a standalone model source for AIKit's current GGUF/llama.cpp image path; runtime-specific LoRA loading must be configured outside that path.
```
[the remark this answers is not in the document]
Comment (sozercan) [reply website/docs/specs-finetune.md]: Fixed in 2992e83 together with the provenance change. The documented tuple is now the exact immutable repository and revision passed to the trainer; the docs also disclose that loadIn4bit may resolve to a prequantized repository.

File: website/docs/fine-tune.md [hunk 4]
```diff
@@ -369,24 +370,55 @@ docker buildx build --builder aikit-builder \
...
+_output/
+└── aikit-adapter/
+    ├── adapter_config.json
+    ├── adapter_model.safetensors
+    ├── tokenizer_config.json
+    ├── tokenizer.json
+    └── optional tokenizer assets and templates
+```
+
+The adapter bundle intentionally excludes both GGUF and base-model weights. Its `adapter_config.json` records the resolved base repository and immutable revision. Load that compatible base separately in a PEFT-capable consumer, then apply the adapter and bundled tokenizer configuration. Architecture, target modules, tokenizer behavior, and the runtime's LoRA support must all be compatible; a similarly named or newer base revision is not assumed compatible.
```
[the remark this answers is not in the document]
Comment (sozercan) [reply website/docs/fine-tune.md]: Fixed in 2992e83. The implementation now records the exact immutable repository and revision loaded for training, and the guide explicitly notes that loadIn4bit may make this a prequantized repository.

File: pkg/aikit/config/finetune_specs.go [hunk 5]
```diff
@@ -242,8 +258,45 @@ type rawFineTuneConfigUnslothSpec struct {
...
+
+	*o = rawFineTuneOutputSpec{
+		Format:   fields.Format,
+		Quantize: fields.Quantize,
+		Name:     fields.Name,
+	}
+	for _, field := range rawFields {
+		fieldName, ok := field.Key.(string)
+		if ok && fieldName == "quantize" {
+			o.quantizeConfigured = true
```
[the remark this answers is not in the document]
Comment (sozercan) [reply pkg/aikit/config/finetune_specs.go]: Fixed in 5396073 and hardened in d32e4a9. Presence is now detected after YAML merge resolution with a sentinel-backed typed presence scan, so inherited values and inherited/direct nulls are treated as explicitly configured without tightening permissive unknown-field parsing. Regression coverage includes merged values/nulls, direct overrides, scalar aliases, merge-sequence precedence, and duplicate-key behavior.

**Comment by sozercan** [comment]
@codex review

**Comment by sozercan** [comment]
@codex review
